src/contract/aci.ts

```diff
@@ -0,0 +1,516 @@
+  * @param params Contract init function arguments array
+  * @param options
+  * @returns deploy info
+  */
+  instance.deploy = async (params = [], options: object): Promise<ContractInstance['deployInfo']> => {
+    const opt = { ...instance.options, ...options }
+    if (instance.bytecode == null) await instance.compile(opt)
+    if (opt.callStatic === true) return await instance.call('init', params, opt)
+    if (instance.deployInfo.address != null) throw new DuplicateContractError()
+
+    const ownerId: string = await onAccount.address(opt)
+    // TODO onAccount shouldn't have .buildTx method
+    const tx = await onAccount.buildTx(TX_TYPE.contractCreate, {
+      ...opt,
+      gasLimit: opt.gasLimit ?? await instance._estimateGas('init', params, opt),
+      callData: instance.calldata.encode(instance._name, 'init', params),
+      code: instance.bytecode,
+      ownerId
+    })
+    const contractId = buildContractIdByContractTx(tx)
+    const { hash, rawTx, result, txData } = await sendAndProcess(tx, opt)
+    instance.deployInfo = Object.freeze({
+      result,
+      owner: ownerId,
+      transaction: hash,
+      rawTx,
+      txData,
+      address: contractId
+    })
+    return instance.deployInfo
+  }
+
+  /**
+  * Get function schema from contract ACI object
+  * @param name Function name
+  * @return function ACI
+  */
+  function getFunctionACI (name: string): Partial<FunctionACI> {
+    const fn = instance._aci.encodedAci.contract.functions.find(
+      (f: { name: string }) => f.name === name)
+    if (fn != null) {
+      return fn
+    }
+    if (name === 'init') return { payable: false }
+    throw new NoSuchContractFunctionError(`Function ${name} doesn't exist in contract`)
+  }
+
+  /**
+  * Call contract function
+  * @alias module:@aeternity/aepp-sdk/es/contract/aci
+  * @param fn Function name
+  * @param params Array of function arguments
+  * @param options Array of function arguments
+  * @returns CallResult
+  */
+  instance.call = async (fn: string, params: any[] = [], options: object = {}) => {
+    const opt = { ...instance.options, ...options }
+    const fnACI = getFunctionACI(fn)
+    const contractId = instance.deployInfo.address
+
+    if (fn == null) throw new MissingFunctionNameError()
+    if (fn === 'init' && opt.callStatic === false) throw new InvalidMethodInvocationError('"init" can be called only via dryRun')
+    if (contractId == null && fn !== 'init') throw new InvalidMethodInvocationError('You need to deploy contract before calling!')
+    if (fn !== 'init' && opt.amount > 0 && fnACI.payable === false) throw new NotPayableFunctionError(opt.amount, fn)
+
+    const callerId = await onAccount.address(opt).catch((error: any) => {
+      if (opt.callStatic === true) return DRY_RUN_ACCOUNT.pub
+      else throw error
+    })
+    const callData = instance.calldata.encode(instance._name, fn, params)
+
+    let res: any
+    if (opt.callStatic === true) {
+      if (typeof opt.top === 'number') {
```

Comment:
```suggestion
      if (opt.top != null) {
```
we don't have to check its type, just if it present or not